hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java

```diff
@@ -819,4 +832,74 @@ private void connectToFeedCommChannel() throws Exception {
         feedComm = new FeedComm(this.httpClientBuilder, this.configuration, this.feedId, this.dmrServerInventories);
         feedComm.connect();
     }
+
+    /**
+     * A filter that replaces 0.0.0.0 server address with the list of addresses got from
+     * {@link InetAddress#getByName(String)} where the argument of {@code getByName(String)} is the host the agent uses
+     * to query the AS'es DMR.
+     */
+    private static final class ServerAddressResolver implements InventoryStorage {
+        private final InventoryStorage delegate;
+        private final ManagedServer server;
+
+        public ServerAddressResolver(ManagedServer server, InventoryStorage delegate) throws UnknownHostException {
+            super();
+            this.delegate = delegate;
+            this.server = server;
+        }
+
+        private InetAddress[] resolveHost() throws UnknownHostException {
+            String host = null;
+            if (server instanceof RemoteDMRManagedServer) {
+                RemoteDMRManagedServer remoteServer = (RemoteDMRManagedServer) server;
+                host = remoteServer.getHost();
+            } else if (server instanceof LocalDMRManagedServer) {
+                host = InetAddress.getLocalHost().getCanonicalHostName();
+            } else {
+                throw new IllegalStateException("Unexpected type of managed server '" + server.getClass().getName()
+                        + "'; expected '" + RemoteDMRManagedServer.class.getName() + "' or '"
+                        + LocalDMRManagedServer.class.getName() + "'. Please report this bug.");
+            }
+            return InetAddress.getAllByName(host);
+        }
+
+        @Override
+        public void storeResourceType(ResourceType<?, ?, ?, ?> resourceType) {
+            delegate.storeResourceType(resourceType);
+        }
+
+        @Override
+        public void storeResource(Resource<?, ?, ?, ?, ?> resource) {
+            final String WF_TYPE_NAME = "WildFly Server";
+            final String IP_ADDRESSES_PROPERTY_NAME = "Bound Address";
+            String typeName = resource.getResourceType().getName().getNameString();
+            if (WF_TYPE_NAME.equals(typeName) && resource instanceof DMRResource) {
```

Comment:
What if you just checked for a null parent? That would tell you its the top root resource which is the one you want. No need for having it to be named something specific like "WildFly Server" -  for example:

if (resource instanceof DMRResource && resource.getParent() == null)

That way we know its the top root resource without having to know its actual name. I think that would work and would mean you can get rid of WF_TYPE_NAME completely.